src/guides/v2.3/extension-dev-guide/price-adjustments.md

```diff
@@ -0,0 +1,411 @@
+    }
+
+    /**
+     * Extract adjustment amount from the given amount value
+     *
+     * @param float $amount
+     * @param SaleableInterface $saleableItem
+     * @param null|array $context
+     * @return float
+     */
+    public function extractAdjustment($amount, SaleableInterface $saleableItem, $context = [])
+    {
+        $return = $amount - self::ADJUSTMENT_VALUE;
+        return $return;
+    }
+
+    /**
+     * Apply adjustment amount and return result value
+     *
+     * @param float $amount
+     * @param SaleableInterface $saleableItem
+     * @param null|array $context
+     * @return float
+     */
+    public function applyAdjustment($amount, SaleableInterface $saleableItem, $context = [])
+    {
+        $return = $amount + self::ADJUSTMENT_VALUE;
+        return $return;
+    }
+
+    /**
+     * Check if adjustment should be excluded from calculations along with the given adjustment
+     *
+     * @param string $adjustmentCode
+     * @return bool
+     */
+    public function isExcludedWith($adjustmentCode)
+    {
+        return $this->getAdjustmentCode() === $adjustmentCode;
+    }
+
+    /**
+     * Return sort order position
+     *
+     * @return int
+     */
+    public function getSortOrder()
+    {
+        return 21;
+    }
+}
+```
+
+Adjustment logic is defined in `extractAdjustment` and `applyAdjustment` functions.
+
+
+## What Areas Price Adjustments Affect
+
+Price Adjustments affect storefront product prices. 
+
+Price Adjustments **will not** affect quote item prices so when a product is added to the cart any price adjustments defined the way described above are discarded.
+
+
+## Correct Way To Do Price Adjustments for Quote Items
+
+The correct way to do price adjustments for quote items is to add a custom quote total.
+
+Add the following to `VENDOR/MODULE/etc/sales.xml`:
+
+```xml
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Sales:etc/sales.xsd">
+    <section name="quote">
+        <group name="totals">
+            <item name="custom-surcharge" instance="VENDOR\MODULE\Model\Quote\Surcharge" sort_order="1000"/>
+        </group>
+    </section>
+</config>
+```
+
+Then `VENDOR/MODULE/Model/Quote/Surcharge.php`:
+
+```php
+namespace VENDOR\MODULE\Model\Quote;
+
+class Surcharge extends \Magento\Quote\Model\Quote\Address\Total\AbstractTotal
+{
+
+   const COLLECTOR_TYPE_CODE = 'custom-surcharge';
+
+   /**
+    * Custom constructor.
+    */
+   public function __construct()
+   {
+       $this->setCode(self::COLLECTOR_TYPE_CODE);
+
+   }
+
+   /**
+    * Collect address discount amount
+    *
+    * @param \Magento\Quote\Model\Quote $quote
+    * @param \Magento\Quote\Api\Data\ShippingAssignmentInterface $shippingAssignment
+    * @param \Magento\Quote\Model\Quote\Address\Total $total
+    * @return $this
+    */
+   public function collect(
+        \Magento\Quote\Model\Quote $quote,
```

Comment:
Lets change those in more correct way by moving \Magento\Quote\Model\Quote into "use" section of the class.
Same everywhere